### PR TITLE
feat: EMR Serverless

### PR DESCRIPTION
**Pull request by kukushking**
Add initial integration with EMR Serverless. This extends beyond boto3 calls with some defaults & ability to kick off and wait for the job to finish. Open to suggestions for other potential features.

### Feature or Bugfix
- Feature

### Detail
- Add create_application wrapper with defaults
- Move _get_unique_suffix to test util
- Add type overload for EMR serverelsss boto3 client
- Add create_application test case
- Add run_job and wait_job
- Docstrings & minor fixes

### Todo
- Add ability to provide SQL query directly to the Hive job
- ...

### Relates
- #2242 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubCodeBuild8756EF16-4rfo0GHQ0u9a
* Commit ID: 23e0be57e2abfced2c10630b047911a0048d7a94
* Result: FAILED
* [Build Logs](https://gqgcj9s2gc.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=529f5e95-ae7f-4519-b259-357ce58622b8%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubDistributedCodeBuild6-jWcl5DLmvupS
* Commit ID: 23e0be57e2abfced2c10630b047911a0048d7a94
* Result: SUCCEEDED
* [Build Logs](https://p0rshq9q5i.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=c48115d6-7454-4938-8fd0-2abb1d8c8631%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by kukushking** [comment]
Re-running the tests.

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubCodeBuild8756EF16-4rfo0GHQ0u9a
* Commit ID: bf0a05e0e13db521971ec8d6986cc9585eab4621
* Result: SUCCEEDED
* [Build Logs](https://gqgcj9s2gc.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=cf9a0059-c31b-4de3-b297-796668d3c7d5%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubDistributedCodeBuild6-jWcl5DLmvupS
* Commit ID: bf0a05e0e13db521971ec8d6986cc9585eab4621
* Result: SUCCEEDED
* [Build Logs](https://p0rshq9q5i.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=b297cdbd-47b9-40c0-b835-3666194dc882%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

File: awswrangler/emr_serverless.py [hunk 1]
```diff
@@ -0,0 +1,272 @@
...
+    if job_type == "Spark":
+        job_args["jobDriver"] = {
+            "sparkSubmit": job_driver_args,
+        }
+    elif job_type == "Hive":
+        job_args["jobDriver"] = {
+            "hive": job_driver_args,
+        }
+    else:
+        raise ValueError(f"Unknown job type `{job_type}`")
```
Comment (LeonLuttenberger) [inline awswrangler/emr_serverless.py:204]: Maybe throw `exceptions.InvalidArgumentValue` instead?

File: pyproject.toml [hunk 2]
```diff
@@ -84,7 +84,7 @@ wheel = "^0.38.1"
 
 # Lint
 black = "^23.1.0"
-boto3-stubs = {version = "1.26.47", extras = ["athena", "chime", "cloudwatch", "dynamodb", "ec2", "emr", "glue", "kms", "lakeformation", "logs", "neptune", "opensearch", "opensearchserverless", "quicksight", "rds", "rds-data", "redshift", "redshift-data", "s3", "secretsmanager", "ssm", "sts", "timestream-query", "timestream-write"]}
```
Comment (LeonLuttenberger) [inline pyproject.toml:87]: The `poetry.lock` file is out sync, so you should run `poetry lock --no-update`

File: awswrangler/_utils.py [hunk 3]
```diff
@@ -334,6 +334,16 @@ def client(
     ...
 
 
+@overload
+def client(
+    service_name: 'Literal["emr-serverless"]',
+    session: Optional[boto3.Session] = None,
+    botocore_config: Optional[Config] = None,
+    verify: Optional[Union[str, bool]] = None,
+) -> "EMRServerlessClient":
```
Comment (LeonLuttenberger) [inline awswrangler/_utils.py:343]: It's not imported :)

File: awswrangler/emr_serverless.py (hunk 1 above)
Comment (kukushking) [reply awswrangler/emr_serverless.py]: Done, thanks

File: pyproject.toml (hunk 2 above)
Comment (kukushking) [reply pyproject.toml:87]: That's right, thanks

File: awswrangler/_utils.py (hunk 3 above)
Comment (kukushking) [reply awswrangler/_utils.py:344]: Ahhhh, that's why mypy wouldn't pick it up. Thanks!

**Review by LeonLuttenberger — APPROVED** [review]

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubCodeBuild8756EF16-4rfo0GHQ0u9a
* Commit ID: b6f1b57ed9461c6ff6ffe18639ccb86e7e05b4e0
* Result: SUCCEEDED
* [Build Logs](https://gqgcj9s2gc.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=07056a21-4e87-4054-8ded-e031c8cacc5f%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubDistributedCodeBuild6-jWcl5DLmvupS
* Commit ID: b6f1b57ed9461c6ff6ffe18639ccb86e7e05b4e0
* Result: SUCCEEDED
* [Build Logs](https://p0rshq9q5i.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=e2e87328-91e2-4872-b138-4698fd15ccc7%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubDistributedCodeBuild6-jWcl5DLmvupS
* Commit ID: d80fc856be4d04f174010965c5359acacf13b091
* Result: FAILED
* [Build Logs](https://p0rshq9q5i.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=2a2bd978-48d4-4515-8e5a-b7b78eedf8af%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubCodeBuild8756EF16-4rfo0GHQ0u9a
* Commit ID: d80fc856be4d04f174010965c5359acacf13b091
* Result: FAILED
* [Build Logs](https://gqgcj9s2gc.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=270b956f-4771-4e50-9fe6-1f8577d96e0a%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubCodeBuild8756EF16-4rfo0GHQ0u9a
* Commit ID: f4dc2ad09fbe77aa250729cf4a4181e15c3d91ea
* Result: FAILED
* [Build Logs](https://gqgcj9s2gc.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=f097a391-09b1-44f8-8cca-ed18112ef5a2%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubDistributedCodeBuild6-jWcl5DLmvupS
* Commit ID: f4dc2ad09fbe77aa250729cf4a4181e15c3d91ea
* Result: FAILED
* [Build Logs](https://p0rshq9q5i.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=7195679d-6c30-4607-838f-f0af0cbe6b58%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Review by jaidisido — APPROVED** [review]

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubCodeBuild8756EF16-4rfo0GHQ0u9a
* Commit ID: 442bc8eab141e811533eb526eca0a6c571710312
* Result: SUCCEEDED
* [Build Logs](https://gqgcj9s2gc.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=e1cd02e3-a677-4bbc-b787-deb3c9d0b6e3%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubCodeBuild8756EF16-4rfo0GHQ0u9a
* Commit ID: e471231bfc91d88acb0b6abc7da4c8634d2d0a63
* Result: SUCCEEDED
* [Build Logs](https://gqgcj9s2gc.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=833d69c6-f995-4d8d-bc3b-10e32f7293aa%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubDistributedCodeBuild6-jWcl5DLmvupS
* Commit ID: 442bc8eab141e811533eb526eca0a6c571710312
* Result: SUCCEEDED
* [Build Logs](https://p0rshq9q5i.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=521c8cfc-8920-4e42-950e-3623bbe1eeaf%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubDistributedCodeBuild6-jWcl5DLmvupS
* Commit ID: e471231bfc91d88acb0b6abc7da4c8634d2d0a63
* Result: SUCCEEDED
* [Build Logs](https://p0rshq9q5i.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=8d3ac627-f14a-4f60-be09-ce614ccb1d5a%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Pull request merged**